frontend/src/container/OrganizationSettings/InviteTeamMembers/index.tsx

```diff
@@ -1,107 +1,81 @@
-		setAllMembers((state) => [
-			...state,
-			{
-				email: '',
-				name: '',
-				role: 'VIEWER',
-			},
-		]);
-	};
-
-	const onChangeHandler = useCallback(
-		(value: string, index: number, type: string): void => {
-			setAllMembers((prev) => [
-				...prev.slice(0, index),
-				{
-					...prev[index],
-					[type]: value,
-				},
-				...prev.slice(index, prev.length - 1),
-			]);
-		},
-		[setAllMembers],
-	);
-
 	return (
 		<>
 			<TitleWrapper>
 				<Typography>{t('email_address')}</Typography>
 				<Typography>{t('name_optional')}</Typography>
 				<Typography>{t('role')}</Typography>
 			</TitleWrapper>
-			<Form>
-				<Space direction="vertical" align="center" size="middle">
-					{allMembers.map((e, index) => (
-						<Space key={Number(index)} direction="horizontal">
-							<Input
-								placeholder={t('email_placeholder')}
-								value={e.email}
-								onChange={(event): void => {
-									onChangeHandler(event.target.value, index, 'email');
-								}}
-								required
-							/>
-							<Input
-								placeholder={t('name_placeholder')}
-								value={e.name}
-								onChange={(event): void => {
-									onChangeHandler(event.target.value, index, 'name');
-								}}
-								required
-							/>
-							<SelectDrawer
-								value={e.role}
-								onSelect={(value: unknown): void => {
-									if (typeof value === 'string') {
-										onChangeHandler(value, index, 'role');
-									}
-								}}
-							>
-								<Option value="ADMIN">ADMIN</Option>
-								<Option value="VIEWER">VIEWER</Option>
-								<Option value="EDITOR">EDITOR</Option>
-							</SelectDrawer>
-						</Space>
-					))}
-					<Button onClick={onAddHandler} icon={<PlusOutlined />} type="default">
-						{t('add_another_team_member')}
-					</Button>
-				</Space>
+			<Form
+				form={form}
+				onFinish={onFinish}
+				initialValues={{ members: [{ email: '', name: '', role: 'VIEWER' }] }}
+			>
+				<Form.List name="members">
+					{(fields, { add }): JSX.Element => (
+						<SpaceContainer direction="vertical" align="center" size="middle">
+							{fields.map(({ key, name }) => (
+								<Space key={key} direction="horizontal">
+									<Form.Item
+										name={[name, 'email']}
+										rules={[{ required: true, message: 'Missing email' }]}
+									>
+										<Input placeholder={t('email_placeholder')} />
+									</Form.Item>
+									<Form.Item
+										name={[name, 'name']}
+										rules={[{ required: true, message: 'Missing name' }]}
+									>
+										<Input placeholder={t('name_placeholder')} />
+									</Form.Item>
+									<Form.Item name={[name, 'role']} initialValue="VIEWER">
+										<SelectDrawer>
+											<Option value="ADMIN">ADMIN</Option>
+											<Option value="VIEWER">VIEWER</Option>
+											<Option value="EDITOR">EDITOR</Option>
+										</SelectDrawer>
+									</Form.Item>
+								</Space>
+							))}
+							<Form.Item>
+								<Button
+									onClick={(): void => add()}
```

Comment:
```suggestion
									onClick={add}
```

Reply:
Isn't this gonna complain about that eslint error - no function return type?

Reply:
nopes

Reply:
Okay, resolved